### DISCUSSION TITLE
ERR_PNPM_UNSUPPORTED_ENGINE: Renovatebot incompatible with our local pnpm version

### DISCUSSION THREAD

**Discussion by samydoesit** [renovatebot/renovate#23731] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.34.0 - latest 

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Cloud (bitbucket.org)

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

I recently noted that Renovatebot switched to using pnpm as its package manager. However, when running the config-validator of Renovatebot with our local package.json's `engines` constraint for `pnpm`, the validation fails with the following error `ERR_PNPM_UNSUPPORTED_ENGINE`.

- We are using pnpm version `8.6.5`.
- We have set `engine-strict="true"` in `.npmrc`
- It seems Renovatebot is on pnpm version `8.6.11`.
- Downgrading to Renovate `36.31.0` serves as a temporary workaround for us.

### Steps to reproduce
Reproduction: https://github.com/samydoesit/repro-renovate-pnpm-engine-strict
Install: `pnpm install`
Run validateRenovate: `pnpm run validateRenovate`

### Relevant debug logs

```bash
ERR_PNPM_UNSUPPORTED_ENGINE  Unsupported environment (bad pnpm and/or Node.js version)

This error happened while installing a direct dependency of /tmp/directory/.pnpm-store/v3/tmp/dlx-468
Your pnpm version is incompatible with "registry.npmjs.org/renovate/36.34.0".

Expected version: ^8.6.11
Got: 8.6.5
```

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment]
I suggest running the validator:
- Without `pnpm run`
- Without `pnpm` (if above fails)

**Reply by samydoesit** [reply]
Okay, I had no luck using `pnpm`. I tried `npx -p renovate renovate-config-validator`, and this seemed to work. However, it definitely worked with `pnpm` before.

**Reply by rarkins** [reply]
Maybe you can run pnpm with "ignore engines"

**Reply by samydoesit** [reply]
Unfortunately, i can't find such an option in pnpm.

**Reply by rarkins** [reply]
@zkochan any ideas?

**Reply by zkochan** [reply]
I guess you should remove this field: 

https://github.com/samydoesit/repro-renovate-pnpm-engine-strict/blob/b3ee4aafe435696df93d0c7807f2e3605a3090a1/package.json#L10

**Reply by samydoesit** [reply]
Removing the `pnpm` field from the `engines` section doesn't resolve the issue, i still receive the same error.

**Reply by rarkins** [reply]
I wonder if we should delete the engines.pnpm field for our package.json field before publishing

**Reply by zkochan** [reply]
The `engines.pnpm` field of the installed packages should be ignored. If it is not, then it is a bug

**Reply by viceice** [reply]
maybe we can lower our engines constraints for pnpm?

@zkochan which version is safe to downgrade to?

**Reply by zkochan** [reply]
You use an exact version in `engines.pnpm`, you can't expect renovate to always much it. At least, use a range.

**Reply by viceice** [reply]
we've a range

https://github.com/renovatebot/renovate/blob/6c42022f78c9ff73983e95c97099bb0cc4442b27/package.json#L138

I asked to lower that to allow older pnpm versions. so just use `^8.0.0` ? 🤔

**Reply by zkochan** [reply]
That should suffice

**Reply by viceice** [reply, marked as answer]
ok, this should fix the issue:
- #23748
